app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarUIView.kt

```diff
@@ -37,56 +44,179 @@ class AwesomeBarUIView(
-                    loadUrlUseCase,
-                    components.utils.icons
-                )
+            shortcutsEnginePickerProvider =
+                    ShortcutsSuggestionProvider(
+                        components.search.searchEngineManager,
+                        this,
+                        ::selectShortcutEngine,
+                        ::selectShortcutEngineSettings)
+        }
+    }
+
+    private fun showShortcutEnginePicker() {
+        with(container.context) {
+            search_shortcuts_button.background = getDrawable(R.drawable.search_pill_background)
+            search_shortcuts_button.compoundDrawables[0].setTint(ContextCompat.getColor(this,
+                DefaultThemeManager.resolveAttribute(R.attr.pillWrapperBackground, this)))
+            search_shortcuts_button.textColor = ContextCompat.getColor(this,
+                DefaultThemeManager.resolveAttribute(R.attr.pillWrapperBackground, this))
+
+            view.removeAllProviders()
+            view.addProviders(shortcutsEnginePickerProvider!!)
+        }
+    }
+
+    private fun hideShortcutEnginePicker() {
+        with(container.context) {
+            search_shortcuts_button.setBackgroundColor(ContextCompat.getColor(this,
+                DefaultThemeManager.resolveAttribute(R.attr.pillWrapperBackground, this)))
+            search_shortcuts_button.compoundDrawables[0].setTint(ContextCompat.getColor(this,
+                DefaultThemeManager.resolveAttribute(R.attr.searchShortcutsTextColor, this)))
+            search_shortcuts_button.textColor = ContextCompat.getColor(this,
+                DefaultThemeManager.resolveAttribute(R.attr.searchShortcutsTextColor, this))
+
+            view.removeProviders(shortcutsEnginePickerProvider!!)
+        }
+    }
+
+    private fun showSuggestionProviders() {
+        if (Settings.getInstance(container.context).showSearchSuggestions()) {
+            view.addProviders(searchSuggestionProvider!!)
+        }
+
+        view.addProviders(
+            clipboardSuggestionProvider!!,
+            historyStorageProvider!!,
+            sessionProvider!!
+        )
+    }
+
+    private fun selectShortcutEngine(engine: SearchEngine) {
+        actionEmitter.onNext(AwesomeBarAction.SearchShortcutEngineSelected(engine))
+    }
+
+    private fun setShortcutEngine(engine: SearchEngine) {
+        with(container.context) {
+
+            val draw = getDrawable(R.drawable.ic_search)
+            draw?.setTint(ContextCompat.getColor(this, R.color.search_text))
+
+            searchSuggestionFromShortcutProvider = SearchSuggestionProvider(
+                components.search.searchEngineManager.getDefaultSearchEngine(this, engine.name),
+                shortcutSearchUseCase,
+                components.core.client,
+                mode = SearchSuggestionProvider.Mode.MULTIPLE_SUGGESTIONS
+                // TODO icon = draw?.toBitmap()
             )
         }
     }
 
+    private fun showSearchSuggestionProvider() {
```

Comment:
Is it bad that all this logic for showing the search suggestion provider / shortcut engine settings exists inside of `AwesomeBarUIView`? If so, what where you recommend I move it?